### PR TITLE
[core] [integrations] Removes checks and config

### PR DESCRIPTION
**Pull request by gmmeyer**
I apologize for the size of this Pull Request. It's gigantic. It's so large that Github starts getting glitchy when I try to view it here.

Here's what it does, in general terms:

First, this removes nearly everything. It removes almost all checks except hdfs, mesos and jenkins because they are deprecated, and kubernetes, docker and agent metrics because they are tied too closely to the agent itself. It removes all config files except the ones for those checks and the same for the autoconf files.

Second, a bunch of existing tests do things like loading existing checks or existing yaml files. 

Unfortunately, they specifically check for the check they're loading, checking to see whether or not the "disk" check is running. The way I think is best to fix this, because it adds the least complexity, is to add them to fixtures, move them to the proper place when the test runs, and then remove them afterwards.

This is also the solution that is least confusing, as these checks are no longer properly part of the codebase of the agent except as fixtures for the tests.

Third, this updates the readme and contributions files.

Fourth, this removes the relevant rakefiles and the commands from Travis.

I think that this about covers everything, however, if I have forgotten anything, let me know and I will add it back in here.

### CODE REVIEW

**Comment by gmmeyer** [comment]
@degemer It's also used by service discovery. I didn't think of modifying the path, do you think that would be less disruptive?

I can remove them, yes.

**Comment by gmmeyer** [comment]
I think that this still adds less complexity than that option, but am open to the discussion!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,6 +28,11 @@ Your Pull Request **must** always pass the Travis/Appveyor tests before being me
 
 _If you are adding a dependency (python module, library, ...), please check the [corresponding section](#add-dependencies)._
 
+## [Integrations](https://github.com/DataDog/integrations-core)
+
+All integrations, except for the kubernetes and docker ones, have been moved to the [Integration SDK](https://github.com/DataDog/integrations-core). Please look there to contribute to them.
```
Comment (irabinovitch) [inline CONTRIBUTING.md]: Are we planning to move docker and k8s as well?

File: README.md [hunk 2]
```diff
@@ -27,6 +23,10 @@ If you do **not** agree with the licensing terms and wish to contribute code non
 please email us at <info@datadoghq.com> before submitting your
 pull request.
 
+# [Integration SDK](https://github.com/DataDog/integrations-core)
+
+All integrations, except for the kubernetes and docker ones, have been moved to the [Integration SDK](https://github.com/DataDog/integrations-core). Please look there to see and contribute to any Integrations.
```
Comment (irabinovitch) [inline README.md]: All integrations, except for the kubernetes and docker, have been moved to the [Integration SDK](https://github.com/DataDog/integrations-core). Please look there to submit related issues, PRs, or review the latest changes.

File: CONTRIBUTING.md (hunk 1 above)
Comment (gmmeyer) [inline CONTRIBUTING.md]: Oh I need to update that as well, we have the deprecated hdfs, jenkins and mesos checks in here too, along with the agent_metrics check.

We are but not before the release. Currently they are too tied into the core agent to remove.

Comment (irabinovitch) [inline CONTRIBUTING.md]: Lets put that in their individual README pages too?

Comment (gmmeyer) [inline CONTRIBUTING.md]: Individual readme pages?

File: tests/checks/fixtures/checks/disk.py [hunk 3]
```diff
@@ -43,6 +43,9 @@ def check(self, instance):
         # Windows and Mac will always have psutil
         # (we have packaged for both of them)
         if self._psutil():
+            if Platform.is_linux():
```
Comment (truthbk) [inline tests/checks/fixtures/checks/disk.py]: What was this fixture?

File: conf.d/auto_conf/apache.yaml [hunk 4]
```diff
@@ -1,7 +0,0 @@
-docker_images:
```
Comment (truthbk) [inline conf.d/auto_conf/apache.yaml]: we have to add these to integrations-core....

Comment (truthbk) [inline conf.d/auto_conf/apache.yaml]: https://github.com/DataDog/integrations-core/pull/216

File: tests/checks/fixtures/checks/disk.py (hunk 3 above)
Comment (truthbk) [inline tests/checks/fixtures/checks/disk.py]: Got it, for service discovery.

**Pull request merged**